### ISSUE TITLE
Installation fail due to charset

### ISSUE THREAD

**Issue by michelpf** [aerkalov/ebooklib#132]
Could you fix the charmap, please? The latest version is OK, but the 0.15, required by texetract is not.

Collecting EbookLib==0.15 (from textract)
  Using cached EbookLib-0.15.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\MICHEL~1.FER\AppData\Local\Temp\pip-build-rtyxiy1a\EbookLib\setup.py", line 13, in <module>
        long_description = open('README.md').read(),
      File "c:\python\python36\lib\encodings\cp1252.py", line 23, in decode
        return codecs.charmap_decode(input,self.errors,decoding_table)[0]
    UnicodeDecodeError: 'charmap' codec can't decode byte 0x8d in position 1671: character maps to <undefined>

**Comment by isnok** [comment]
Now I get this similar error (from `textract==1.6.1`):

```
Collecting EbookLib==0.15 (from textract==1.6.1->-r code/requirements.txt (line 3))
  Using cached EbookLib-0.15.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-v6_q9shv/EbookLib/setup.py", line 13, in <module>
        long_description = open('README.md').read(),
      File "/venv/lib/python3.5/encodings/ascii.py", line 26, in decode
        return codecs.ascii_decode(input, self.errors)[0]
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xc5 in position 1658: ordinal not in range(128)
```

Strangely it works on my regular OS (debian/sid) with `python 3.5.4`, but i get this error in the official "debian" docker container with `python 3.5.3`. I'm wondering if i might have some environment variables set on my host, that change the way how README.md is processed.

**Comment by isnok** [comment]
I found the reason:

As described in this [stackoverflow thread](https://stackoverflow.com/questions/23917729/switching-to-python-3-causing-unicodedecodeerror#23917799), `locale.getpreferredencoding(False)` returns something different on my host (`UTF-8`) than in the docker container (`ANSI_X3.4-1968`).
I think the best way to solve this would be to specify an encoding while reading the file (or reading it binary, and decode manually with specified encoding). Hope this helps.

For now, I will figure out how to set that locale setting in my container.

**Comment by tantale** [comment]
You are right. But the encoding should be set in the `read` function.

The `README.md` file use utf-8 encoding.
On Python 2, if you don't specify and encoding, you read the file using ASCII charset. Which is wrong.

You need to fix like this:

    import io
    
    def read(path):
        with io.open(path, mode="r", encoding="utf-8") as fd:
            return fd.read()

and:

    long_description = read('README.md')

I'll add a pull request for that.

**Comment by psvrao** [comment]
I am facing the same issue while trying to install textract using pip on windows 10, with python 3.5 anaconda distribution. Tried python 3.6 also, both fail with same error.

any suggestions?

```
C:\Users\P Srinivasa Rao>pip install textract
Collecting textract
  Using cached textract-1.6.1.tar.gz
Collecting argcomplete==1.8.2 (from textract)
  Using cached argcomplete-1.8.2-py2.py3-none-any.whl
Collecting chardet==2.3.0 (from textract)
  Using cached chardet-2.3.0-py2.py3-none-any.whl
Collecting python-pptx==0.6.5 (from textract)
  Using cached python-pptx-0.6.5.tar.gz
Collecting docx2txt==0.6 (from textract)
  Using cached docx2txt-0.6.tar.gz
Collecting beautifulsoup4==4.5.3 (from textract)
  Using cached beautifulsoup4-4.5.3-py3-none-any.whl
Collecting xlrd==1.0.0 (from textract)
  Using cached xlrd-1.0.0-py3-none-any.whl
Collecting EbookLib==0.15 (from textract)
  Using cached EbookLib-0.15.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\Public\Documents\Wondershare\CreatorTemp\pip-build-oivoo009\EbookLib\setup.py", line 13, in <module>
        long_description = open('README.md').read(),
      File "C:\Anaconda3\lib\encodings\cp1252.py", line 23, in decode
        return codecs.charmap_decode(input,self.errors,decoding_table)[0]
    UnicodeDecodeError: 'charmap' codec can't decode byte 0x8d in position 1671: character maps to <undefined>

```

**Comment by tantale** [comment]
Sorry, but we can't reinvent the past ;-)

This bug is fixed in EbookLib v0.16, not in v0.15. I think you need to wait the next release of textract, because the version number is corrected in [requirements](https://github.com/deanmalmgren/textract/blob/master/requirements/python) file.
